static/js/search.js

```diff
@@ -68,31 +68,37 @@
         else return "";
     }
 
-    if (getCookie("is_china") === "") {
-        $.ajax({
-            url: "https://ipinfo.io?token=796e43f4f146b1",
-            dataType: "jsonp",
-            success: function (response) {
-                if (response.country == 'CN') {
-                    window.renderPageFindSearchResults()
-                    document.cookie = "is_china=true;" + path + expires
-                } else {
-                    window.renderGoogleSearchResults()
-                    document.cookie = "is_china=false;" + path + expires;
-                }
-            },
-            error: function () {
+    async function checkBlockedSite(url) {
+        try {
+            const response = await fetch(url, { method: 'HEAD', mode: 'no-cors' });
+            // If we reach this point, the site is accessible (since mode: 'no-cors' doesn't allow us to check response.ok)
+            return false;
```

Comment:
What's the timeout on this fetch?

Reply:
I'm in Indonesia at the moment and tried this method on a couple sites that are blocked here, Reddit is one of them. It seemed to take about 18s to timeout before serving the PageFind results. Of course, this test only happens once and then the cookie is set so it won't try again for subsequent searches. 

Still, perhaps there is a way to set a shorter timeout. I will investigate.

Reply:
I've found a way to set a 5s timeout so the longest a user would have to wait is 5s before we can conclude that google is not accessible. After this initial request, the user would not have to wait for searches since the cookie is set.